clamor/gateway/emitter.py

```diff
@@ -1,28 +1,199 @@
+        return func(self, listener)
+
+    return wrapper
+
+
+class Priority(Enum):
+    BEFORE = auto()
+    NORMAL = auto()
+    AFTER = auto()
+
+
+class ListenerPod:
+    """Event listener module
+
+    Listeners that all follow a certain event will exist in the same pod. Pods will separate
+    listeners into self-explanatory categories, before, normal, and after. The listeners will
+    trigger in order from before to after, with each listener triggering synchronously with
+    listeners from the same category.
+
+    Attributes
+    ----------
+    before : set
+        Listeners that will trigger 1st
+    normal : set
+        Listeners that will trigger 2nd
+    after : set
+        Listeners that will trigger 3rd
+    """
+
+    def __init__(self):
+        self.before = set()
+        self.normal = set()
+        self.after = set()
+
+    def __bool__(self):
+        return bool(self.before) or bool(self.normal) or bool(self.after)
+
+    @check_coroutine
+    def add_before(self, listener: Callable[..., Coroutine[Any, Any, None]]):
+        """Add listener (Before)
+
+        Add a coroutine to the before category
+
+        Parameters
+        ----------
+        listener : Coroutine
+            A coroutine to be triggered on it's respective event
+        """
+        self.before.add(listener)
+
+    @check_coroutine
+    def add_normal(self, listener: Callable[..., Coroutine[Any, Any, None]]):
+        """Add listener (Normal)
+
+        Add a coroutine to the normal category
+
+        Parameters
+        ----------
+        listener : Coroutine
+            A coroutine to be triggered on it's respective event
+        """
+        self.normal.add(listener)
+
+    @check_coroutine
+    def add_after(self, listener: Callable[..., Coroutine[Any, Any, None]]):
+        """Add listener (After)
+
+        Add a coroutine to the after category
+
+        Parameters
+        ----------
+        listener : Coroutine
+            A coroutine to be triggered on it's respective event
+        """
+        self.after.add(listener)
+
+    async def emit(self, data: Dict[str, Any]):
+        """Trigger listeners in the pod
+
+        All listeners in the before category will be spawned with the appropriate payload, and
+        once all those have finished, the normal category is triggered, and then the after category.
+
+        Parameters
+        ----------
+        data : dict
+            The payload provided by discord to be distributed
+        """
+        async with create_task_group() as tg:
+            for listener in self.before:
+                await tg.spawn(listener, data)
+        async with create_task_group() as tg:
+            for listener in self.normal:
+                await tg.spawn(listener, data)
+        async with create_task_group() as tg:
+            for listener in self.after:
+                await tg.spawn(listener, data)
 
 
 class Emitter:
```

Comment:
Looks fine. Please fix _on_dispatch in a way that it works again. Including saving the session_id

Reply:
Also fixed, session ID doesn't need to be and isn't an int so I resolved that as well.